### PR TITLE
Overhaul

### PR DESCRIPTION
**Pull request by jhunkeler**
* Unlocked `numpy` (i.e. stop using 'x.x' specifier; it's unlikely this will cause problems)
* Applied jinja2 variables to all recipes 
* Improved package summaries wherever possible
* Removed unused packages
* Bump build revision numbers for each recipe we disabled a numpy lock.
* Bump all pipeline meta-package versions so they will no longer depend on the numpy locked packages. (i.e. all meta-package version are now at `2.0.0`)

Remaining tasks:

- [ ] Add numpy floor versions to each package. (Do we care about this?)

### CODE REVIEW

File: pyds9/meta.yaml [hunk 1]
```diff
@@ -1,33 +1,42 @@
...
     license: GPL
     summary: |
         (Supervised master build)
         Python connection to SAOimage DS9 via XPA
+
 build:
-    number: '0'
-    string: {{ GIT_BUILD_STR }}
+    number: {{number}}
+    string: {{GIT_BUILD_STR}}
```
Comment (justincely) [inline pyds9/meta.yaml]: I don't see this defined anywhere - is it passed in during the build process?

File: pyobjc-core/meta.yaml [hunk 2]
```diff
@@ -6,22 +6,26 @@ about:
...
     - setuptools
     - python x.x
     run:
     - python x.x
+
 source:
     fn: pyobjc-{{ version }}.tar.gz
-    url:
-    - https://bitbucket.org/ronaldoussoren/pyobjc/get/pyobjc-{{ version }}.tar.bz2
+    url: https://bitbucket.org/ronaldoussoren/pyobjc/get/pyobjc-{{ version }}.tar.bz2
```
Comment (justincely) [inline pyobjc-core/meta.yaml]: Spaces left around this one

File: pyqtgraph/meta.yaml [hunk 3]
```diff
@@ -1,29 +1,35 @@
+{% set name = 'pyqtgraph' %}
+{% set version = '0.9.11' %}
+{% set number = '1' %}
+
 about:
     home: http://www.pyqtgraph.org/
     license: MIT
-    summary: A pure-Python graphics library for PyQt/PySide (git master build)
+    summary: |
+        (Supervised master build)
```
Comment (justincely) [inline pyqtgraph/meta.yaml]: what does this mean?

File: sphere/meta.yaml [hunk 4]
```diff
@@ -3,29 +3,34 @@
 {% set number = '0' %}
 
 about:
-    home: https://github.com/spacetelescope/sphere
+    home: https://github.com/spacetelescope/{{sphere}}
```
Comment (justincely) [inline sphere/meta.yaml]: should be {{name}}

File: stsci.sphere/meta.yaml [hunk 5]
```diff
@@ -1,29 +1,37 @@
...
     run:
     - astropy >=1.1
     - matplotlib
-    - numpy x.x
+    - numpy
     - python x.x
+
 source:
-    git_url: https://bitbucket.org/stsci_ssb/stsci.sphere
+    git_url: https://bitbucket.org/stsci_ssb/{{name}}.git
```
Comment (justincely) [inline stsci.sphere/meta.yaml]: Not on github?  Shall we file issue to move this over or does this need to live on bitbucket for some reason?

File: appdirs/meta.yaml [hunk 6]
```diff
@@ -1,21 +1,29 @@
...
 requirements:
     build:
     - setuptools
     - python x.x
     run:
     - python x.x
+
 source:
-    fn: appdirs-1.4.0.tar.gz
+    fn: {{name}-{{ version }}.tar.gz
```
Comment (justincely) [inline appdirs/meta.yaml]: {{name}} missing spaces

File: aprio/meta.yaml [hunk 7]
```diff
@@ -24,6 +27,7 @@ requirements:
     - python-daemon
     - docutils
     - python x.x
+
 source:
-    git_tag: {{ tag }}
+    git_tag: {{tag}}
```
Comment (justincely) [inline aprio/meta.yaml]: {{tag}} missing spaces

File: asdf-standard/meta.yaml [hunk 8]
```diff
@@ -21,6 +29,7 @@ requirements:
     - astropy >=1.1
     - setuptools
     - python x.x
+
 source:
-    git_tag: v1.0.0
-    git_url: https://github.com/spacetelescope/asdf-standard
+    git_tag: {{tag}}
```
Comment (justincely) [inline asdf-standard/meta.yaml]: tag again

File: astroimtools/meta.yaml [hunk 9]
```diff
@@ -1,30 +1,39 @@
...
     - cython
     - matplotlib
-    - numpy x.x
+    - numpy
     - python x.x
+
 source:
-    git_tag: v0.1.1
-    git_url: https://github.com/spacetelescope/astroimtools
+    git_tag: {{tag}}
```
Comment (justincely) [inline astroimtools/meta.yaml]: tag

File: asv/meta.yaml [hunk 10]
```diff
@@ -1,24 +1,34 @@
...
     - six
     - sphinx
-    - numpy x.x
+    - numpy
     - python x.x
+
 source:
-    git_tag: v0.1.1
-    git_url: https://github.com/spacetelescope/asv
+    git_tag: {{tag}}
```
Comment (justincely) [inline asv/meta.yaml]: tag

File: cfitsio/meta.yaml [hunk 11]
```diff
@@ -1,13 +1,22 @@
...
-    name: cfitsio
-    version: '3.370'
+    name: {{ name }}
+    version: {{ version }}
+
 source:
-    fn: cfitsio3370.tar.gz
-    url:
-    - ftp://heasarc.gsfc.nasa.gov/software/fitsio/c/cfitsio3370.tar.gz
+    fn: {{ name }}{{version_short}}.tar.gz
```
Comment (justincely) [inline cfitsio/meta.yaml]: version_short

File: drizzle/meta.yaml [hunk 12]
```diff
@@ -16,19 +16,19 @@ build:
     number: {{ number }}
 
 source:
-    git_tag: {{ tag }}
+    git_tag: {{tag}}
```
Comment (justincely) [inline drizzle/meta.yaml]: tag

File: ginga/meta.yaml [hunk 13]
```diff
@@ -1,28 +1,35 @@
...
     - python x.x
     run:
     - astropy >=1.2
-    - numpy x.x
+    - numpy
     - python x.x
+
 source:
-    git_tag: {{ version }}
+    git_tag: {{tag}}
```
Comment (justincely) [inline ginga/meta.yaml]: tag

File: gwcs/meta.yaml [hunk 14]
```diff
@@ -1,27 +1,37 @@
...
     - asdf
     - astropy >=1.1
-    - numpy x.x
+    - numpy
     - python x.x
+
 source:
-    git_tag: v0.6rc1
-    git_url: https://github.com/spacetelescope/gwcs
+    git_tag: {{tag}}
```
Comment (justincely) [inline gwcs/meta.yaml]: tag

File: imexam/meta.yaml [hunk 15]
```diff
@@ -37,12 +37,12 @@ requirements:
     - photutils
     - matplotlib
     - scipy
-    - numpy x.x
+    - numpy
     - python x.x
 
 source:
     git_tag: {{tag}}
```
Comment (justincely) [inline imexam/meta.yaml]: tag

File: iraf/meta.yaml [hunk 16]
```diff
@@ -1,16 +1,23 @@
...
+    version: {{ version }}
+
 source:
-    fn: iraf.tar.gz
+    fn: {{ name }}.tar.gz
     url:
-    - http://ssb.stsci.edu/stripped/linux/iraf.tar.gz [linux]
-    - http://ssb.stsci.edu/stripped/osx/iraf.tar.gz   [osx]
+    - http://ssb.stsci.edu/stripped/linux/{{ name }}.tar.gz [linux]
+    - http://ssb.stsci.edu/stripped/osx/{{iraf}}.tar.gz   [osx]
```
Comment (justincely) [inline iraf/meta.yaml]: {{iraf}} should be {{ name }}

File: jwxml/meta.yaml [hunk 17]
```diff
@@ -1,24 +1,35 @@
...
     run:
-    - numpy x.x
+    - numpy
     - matplotlib
     - python x.x
+
 source:
-    git_tag: v0.1.0
-    git_url: https://github.com/mperrin/jwxml
+    git_tag: {{tag}}
```
Comment (justincely) [inline jwxml/meta.yaml]: tag

File: mosviz/meta.yaml [hunk 18]
```diff
@@ -11,7 +17,7 @@ build:
   number: {{ number }}
 
 source:
-  git_tag: {{ tag }}
+  git_tag: {{tag}}
```
Comment (justincely) [inline mosviz/meta.yaml]: tag

File: photutils/meta.yaml [hunk 19]
```diff
@@ -21,14 +29,16 @@ requirements:
...
+    - numpy
     - scikit-image
     - scipy
     - setuptools
     - python x.x
+
 source:
-    git_tag: v0.3
-    git_url: https://github.com/astropy/photutils
+    git_tag: {{tag}}
```
Comment (justincely) [inline photutils/meta.yaml]: tag

File: poppy/meta.yaml [hunk 20]
```diff
@@ -19,13 +29,14 @@ requirements:
...
     - matplotlib
     - six [py27]
     - enum34 [py27]
     - setuptools
     - python x.x
+
 source:
-    git_tag: v0.5.1
-    git_url: https://github.com/mperrin/poppy
+    git_tag: {{tag}}
```
Comment (justincely) [inline poppy/meta.yaml]: tag

File: pyobjc-framework-quartz/meta.yaml [hunk 21]
```diff
@@ -6,25 +6,29 @@ about:
...
     preserve_egg_dir: 'true'
+
 package:
     name: {{ name }}
     version: {{ version }}
+
 requirements:
     build:
-    - pyobjc-core =={{ version }}
+    - pyobjc-core =={{version }}
```
Comment (justincely) [inline pyobjc-framework-quartz/meta.yaml]: left side spaces missing

File: stginga/meta.yaml [hunk 22]
```diff
@@ -7,28 +7,33 @@ about:
...
     - ginga
     - astropy >=1.2
-    - numpy x.x
+    - numpy
     - scipy
     - python x.x
+
 source:
-    git_tag: {{ tag }}
+    git_tag: {{tag}}
```
Comment (justincely) [inline stginga/meta.yaml]: tag

File: threading2/meta.yaml [hunk 23]
```diff
@@ -1,23 +1,31 @@
...
     run:
     - python x.x
+
 source:
-    fn: threading2-0.2.1.tar.gz
+    fn: {{ name }}-{{ version }}.tar.gz
     md5: 4ffb227b9a0e1f3e2a42f69f92a47bae
-    url:
-    - https://pypi.python.org/packages/source/t/threading2/threading2-0.2.1.tar.gz
+    url: https://pypi.python.org/packages/source/t/{{ name }}/{{ name }}-{[version}}.tar.gz
```
Comment (justincely) [inline threading2/meta.yaml]: version

File: webbpsf/meta.yaml [hunk 24]
```diff
@@ -33,9 +42,11 @@ requirements:
     - python x.x
     - ipywidgets
     - jwxml
+
 source:
-    git_tag: v0.5.1
+    git_tag: {{tag}}
```
Comment (justincely) [inline webbpsf/meta.yaml]: tag

File: jwst_lib/meta.yaml [hunk 25]
```diff
@@ -1,33 +0,0 @@
-about:
```
Comment (jhunkeler) [inline jwst_lib/meta.yaml]: Wasn't supposed to be in the public channel. Any packages that were derived from this recipe were purged from the main channel a while ago.

File: pyqtgraph/meta.yaml (hunk 3 above)
Comment (jhunkeler) [inline pyqtgraph/meta.yaml]: I know we talked about it, but in case anyone doesn't know, the pipe implies a long or multi-line string.

**Comment by jhunkeler** [comment]
@ely I've taken care of the existing problems.

Remaining tasks:

- [ ] Bump build revision numbers for each recipe we disabled a numpy lock.
(Ensures each package effected gets rebuilt)
- [ ] Add numpy floor versions to each package.
(Do we care about this?)
- [ ] Bump all pipeline meta-package versions so they will no longer depend on the numpy locked packages. 
(Sounds like this might constitute a "major" bump to me)

File: gwcs/meta.yaml [hunk 26]
```diff
@@ -1,27 +1,47 @@
...
+
 package:
+<<<<<<< HEAD
+    name: {{ name }}
+    version: {{ version }}
+
+=======
     name: gwcs
     version: 0.7
+>>>>>>> a96a1285ba6cc7b077ba62e2a73d4592dd78003f
```
Comment (justincely) [inline gwcs/meta.yaml]: whoops - merge conflict remaining here.

File: gwcs/meta.yaml [hunk 27]
```diff
@@ -1,27 +1,47 @@
...
+
 source:
+<<<<<<< HEAD
+    git_tag: {{ tag }}
+    git_url: https://github.com/spacetelescope/{{ name }}.git
+
+=======
     git_tag: 0.7
     git_url: https://github.com/spacetelescope/gwcs
+>>>>>>> a96a1285ba6cc7b077ba62e2a73d4592dd78003f
```
Comment (justincely) [inline gwcs/meta.yaml]: ending here

File: stsci.distutils/meta.yaml [hunk 28]
```diff
@@ -1,24 +1,34 @@
+{% set name = 'stsci.distutils' %}
+{% set version = '0.3.8' %}
+{% set number = '1' %}
+
 about:
-    home: https://github.com/embray/stsci.distutils
-    license: GPL
-    summary: stsci.distutils
+    home: https://github.com/spacetelescope/stsci.distutils
```
Comment (justincely) [inline stsci.distutils/meta.yaml]: {{ name }} for home

**Comment by jhunkeler** [comment]
The humanity. Thanks, I'll fix these now ;)

**Comment by jhunkeler** [comment]
Done.

**Pull request merged**